cou/apps/app.py

```diff
@@ -182,7 +276,279 @@ def _get_os_origin(self) -> str:
+        """Add Plan for refresh current channel.
+
+        This function also identify if charm comes from charmstore and in that case,
+        makes the migration.
+        :param plan: Plan to add refresh current channel as sub step.
+        :type plan: UpgradeStep
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool
+        """
+        switch = None
+        description = (
+            f"Refresh '{self.name}' to the latest revision of '{self.expected_current_channel}'"
+        )
+
+        if self.channel == self.next_channel:
+            logger.warning(
+                "App: %s already has the channel set for the next OpenStack version %s",
+                self.name,
+                self.next_os_release,
+            )
+            return
+
+        if self.charm_origin == "cs":
+            description = f"Migration of '{self.name}' from charmstore to charmhub"
+            switch = f"ch:{self.charm}"
+
+        elif self.channel not in (self.expected_current_channel, self.next_channel):
+            description = (
+                f"Changing '{self.name}' channel from: '{self.channel}' "
+                f"to: '{self.expected_current_channel}'"
+            )
+
+        plan.add_step(
+            UpgradeStep(
+                description=description,
+                parallel=parallel,
+                function=async_upgrade_charm,
+                application_name=self.name,
+                channel=self.expected_current_channel,
+                model_name=self.model_name,
+                switch=switch,
+            )
+        )
+
+    def refresh_next_channel(self, plan: UpgradeStep, parallel: bool = False) -> None:
+        """Add plan for refresh to next channel.
+
+        :param plan: Plan to add refresh next channel as sub step.
+        :type plan: UpgradeStep
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool
+        """
+        if self.channel != self.next_channel:
+            plan.add_step(
+                UpgradeStep(
+                    description=f"Refresh '{self.name}' to the new channel: '{self.next_channel}'",
+                    parallel=parallel,
+                    function=async_upgrade_charm,
+                    application_name=self.name,
+                    channel=self.next_channel,
+                    model_name=self.model_name,
+                )
+            )
+
+    def disable_action_managed(self, plan: UpgradeStep, parallel: bool = False) -> None:
+        """Disable action-managed-upgrade to upgrade as "all-in-one" strategy.
+
+        :param plan: Plan to disable action managed upgrade as sub step.
+        :type plan: UpgradeStep
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool
+        """
+        if self.config.get("action-managed-upgrade"):
+            if self.config["action-managed-upgrade"].get("value", False):
```

Comment:
You could skip one if with
```suggestion
            if self.config.get("action-managed-upgrade", {}).get("value", False):
```

Reply:
Done